wasm/wasi/wit/deps/hermes-cardano/event.wit

```diff
@@ -1,88 +1,55 @@
+    /// - `is-mutable` : Flag indicate whether the block is mutable or not.
+    on-cardano-block: func(network: network, block: block, slot: slot, is-mutable: bool);
 }
 
-/// Cardano API Interface - Export ONLY
-interface event-on-txn {
-    use api.{cardano-blockchain-id, cardano-txn};
+interface event-on-rollback {
+    use api.{network, slot};
 
-    /// Triggered when a cardano transaction event fires.
-    ///
-    /// The module must export this interface to use it.
-    ///
+    /// Trigger when a Cardano rollback event is fired.
+    /// 
     /// ## Parameters
-    ///
-    /// - `blockchain` : The blockchain id the block originated from.
-    /// - `slot`       : The slot the transaction is in.
-    /// - `txn-index`  : The offset in the block this transaction is at.
-    /// - `txn`        : The raw transaction data itself.
-    ///
-    /// Returns:
-    ///     Nothing.
     /// 
-    on-cardano-txn: func(blockchain: cardano-blockchain-id, slot: u64, txn-index: u32, txn: cardano-txn);
+    /// - `network` : The Cardano blockchain network where the rollback occurred.
+    /// - `slot`: The Cardano slot number that the rollback rolls to.
+    /// 
+    /// ## Notes
+    /// 
+    /// Rollback events are guaranteed to be fully processed **before** the next
+    /// block event is delivered. The block event delivered immediately after a
+    /// rollback will correspond to the block at the rollback target `slot`.
+    /// This means that rollback processing **does not** need to reset or re-subscribe the blockchain follower.
+    on-cardano-rollback: func(network: network, slot: slot);
 }
 
-/// Cardano API Interface - Export ONLY
-interface event-on-rollback {
-    use api.{cardano-blockchain-id};
+interface event-on-roll-forward {
+    use api.{network, slot};
 
-    /// Triggered when a cardano rollback event fires.
-    ///
-    /// The module must export this interface to use it.
-    ///
+    /// Trigger when a Cardano roll-forward event is fired.
+    /// 
     /// ## Parameters
-    ///
-    /// - `blockchain` : The blockchain id the rollback originated from.
-    /// - `slot`       : The slot the rollback is targeting. (The next block event will be from this slot.)
-    ///
-    /// Returns:
-    ///     Nothing.
     /// 
-    on-cardano-rollback: func(blockchain: cardano-blockchain-id, slot: u64);
+    /// - `network` : The Cardano blockchain network that the roll-forward occurred.
+    /// - `slot` : The Cardano slot number that the roll-forward rolls to.
+    /// 
+    /// ## Notes
+    /// 
+    /// Roll-forward events are guaranteed to be fully processed **before** the next
+    /// block event is delivered. The block event delivered immediately after a
+    /// roll-forward will correspond to the block at the roll-forward target `slot`. 
+    /// This means that roll-forward processing **does not** need to reset or re-subscribe the blockchain follower.
+    on-cardano-roll-forward: func(network: network, slot: slot);
```

Comment:
```suggestion
    on-cardano-roll-forward: func(id: subscription-id, slot: slot);
```

Same as above.  Subscription id should be enough to identify everything about what the roll forward is for.